feat(filter): Add common errors filter for frequently occurring noise

HazAT opened the pull request:
## Summary

Adds a new `common_errors` inbound filter that filters out commonly occurring error messages that are typically noise and not actionable for developers.

This filter is based on analysis of the most frequently configured error message filters across Sentry customers, targeting patterns like:

- **Network errors**: Failed to fetch, Network Error, connection timeouts
- **Browser quirks**: ResizeObserver loop errors, AbortError
- **Third-party SDK errors**: gtag, fbq, HubSpot, Pinterest pixels
- **Mobile WebView errors**: Java bridge errors, iOS webkit handlers
- **Cancelled requests**: Including translations in multiple languages

The filter follows the same pattern as `browser_extensions` - it's a toggle-based filter with hardcoded regex patterns.

## Changes

- New `relay-filter/src/common_errors.rs` module with ~50 error patterns
- Added `CommonErrors` variant to `FilterStatKey`
- Added `common_errors: FilterConfig` to `ProjectFiltersConfig`

ref: https://github.com/getsentry/sentry/pull/104072
ref: https://github.com/getsentry/sentry/pull/104073

HazAT committed:
feat(common-errors): Implement filtering for commonly occurring error messages

This commit introduces a new filter that targets non-actionable error messages, such as network errors and browser quirks, to reduce noise in event logging. The filter is configurable and integrates with the existing filtering system. Additionally, tests have been added to ensure the filter works as intended.

HazAT committed:
remove chunk load errors and add a few others

HazAT committed:
style: Fix formatting

loewenheim commented on relay-filter/src/common_errors.rs:
```diff
@@ -0,0 +1,902 @@
+        r#"(?ix)
+        # Network and fetch errors - these are typically caused by user network issues
+        Failed\sto\sfetch|
+        Network\s[Ee]rror|
```
`[Ee]` should be unnecessary since the regex is already case-insensitive.

HazAT committed:
refactor(common-errors): Simplify error patterns and remove unused tests

This commit refines the error patterns in the common error filtering logic by consolidating and removing redundant entries. It also eliminates outdated test cases related to hydration, database, timeout, and JSON parsing errors, streamlining the test suite for better maintainability.

HazAT committed:
refactor(common-errors): Remove non-error promise rejection tests and clean up error patterns
